dsgrid/config/annual_time_dimension_config.py

```diff
@@ -70,73 +71,70 @@ def convert_dataframe(
         self,
         df,
         project_time_dim,
-        model_years=None,
-        value_columns=None,
+        value_columns: set[str],
+        scratch_dir_context: ScratchDirContext,
         wrap_time_allowed=False,
         data_adjustment=None,
     ):
-        assert model_years is not None
         assert value_columns is not None
         match self.model.measurement_type:
             case MeasurementType.MEASURED:
-                df = self.map_annual_time_measured_to_datetime(df, project_time_dim, model_years)
+                df = self.map_annual_time_measured_to_datetime(df, project_time_dim)
             case MeasurementType.TOTAL:
-                df = self.map_annual_total_to_datetime(
-                    df, project_time_dim, model_years, value_columns
-                )
+                df = self.map_annual_total_to_datetime(df, project_time_dim, value_columns)
             case _:
                 raise NotImplementedError(f"Unhandled: {self.model.measurement_type}")
 
         return df
 
-    def map_annual_time_measured_to_datetime(self, annual_df, dt_dim, model_years):
+    def map_annual_time_measured_to_datetime(
+        self, annual_df: DataFrame, dt_dim: DateTimeDimensionConfig
+    ):
         """Map a dataframe with MeasuredType.MEASURED to DateTime."""
-        time_col = self.get_load_data_time_columns()[0]
-        df = dt_dim.build_time_dataframe(model_years=model_years)
         dt_time_col = dt_dim.get_load_data_time_columns()[0]
-        return df.join(annual_df, on=F.year(dt_time_col) == annual_df[time_col]).drop(time_col)
+        df = dt_dim.build_time_dataframe()
+        with custom_spark_conf(
+            {"spark.sql.session.timeZone": dt_dim.model.datetime_format.timezone.tz_name}
+        ):
+            years = df.withColumn("year", F.year(dt_time_col)).select("year").distinct().collect()
+            if len(years) != 1:
+                raise Exception("Bug: project time has more than one year: {years=}")
+        return df.crossJoin(annual_df)
 
-    def map_annual_total_to_datetime(self, annual_df, dt_dim, model_years, value_columns):
+    def map_annual_total_to_datetime(
+        self, annual_df: DataFrame, dt_dim: DateTimeDimensionConfig, value_columns: set[str]
+    ):
         """Map a dataframe with MeasuredType.TOTAL to DateTime."""
         assert (
             dt_dim.model.time_type == TimeDimensionType.DATETIME
         ), "dt_dim must be datetime type."
         frequency = dt_dim.model.frequency
-        time_col = self.get_load_data_time_columns()[0]
-        dfs = []
-        for model_year in model_years:
-            if self.model.include_leap_day and model_year % 4 == 0:
+        df = dt_dim.build_time_dataframe()
+        dt_time_col = dt_dim.get_load_data_time_columns()[0]
+        with custom_spark_conf(
```

Comment:
I'm not sure that this logic is correct. Do we need to handle the case of a project time range that spans one year?